### PR TITLE
Add support for lazy loading site meta

### PR DESCRIPTION
**Pull request by spacedmonkey**
Trac ticket: https://core.trac.wordpress.org/ticket/58185

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Review by JJJ — APPROVED** [review]

File: tests/phpunit/tests/multisite/siteMeta.php [hunk 1]
```diff
@@ -250,14 +246,84 @@ public function test_query_update_site_meta_cache_true() {
...
+			);
 
+			$this->assertSameSets( array( self::$site_id2 ), $q->sites, 'Site query should return the second test site' );
+
+			get_site_meta( self::$site_id2 );
+
+			$args     = $filter->get_args();
+			$first    = reset( $args );
+			$site_ids = end( $first );
+			$this->assertSameSets( $site_ids, array( self::$site_id, self::$site_id2 ), 'This should have two site\'s meta' );
```
Comment (peterwilsoncc) [inline tests/phpunit/tests/multisite/siteMeta.php]:
```suggestion
			$this->assertSameSets( $site_ids, array( self::$site_id, self::$site_id2 ), 'This should have two sites\' meta' );
```

Nit: sometimes I hate me too :D

File: src/wp-includes/user.php [hunk 2]
```diff
@@ -1004,9 +1004,8 @@ function get_blogs_of_user( $user_id, $all = false ) {
 
 	if ( ! empty( $site_ids ) ) {
 		$args = array(
-			'number'                 => '',
-			'site__in'               => $site_ids,
-			'update_site_meta_cache' => false,
```
Comment (peterwilsoncc) [inline src/wp-includes/user.php:1009]: This will now queue the meta for lazy loading, whereas it wasn't loaded at all in the past.

Why is this change needed/an improvement?

**Review by peterwilsoncc — COMMENTED** [review]
LGTM based on a code read through, tests make sense. I can never get the `blogs` and `site` terminology in code vs UI straight in my head but looking at the table names, it all looks correct.

Added a couple of minor notes inline.

File: src/wp-includes/user.php (hunk 2 above)
Comment (spacedmonkey) [reply src/wp-includes/user.php:1009]: We made this change here https://github.com/WordPress/wordpress-develop/commit/82be24c885e4c69745ae0fdb14177617393a9317#diff-ff818ec19f0f94e4175f72bd2cc4cd67c455c0c50e5b651690a1e312c284ba06. 

Everywhere else WP_Site_Query a single site is loaded and site meta is not specially useful. This results an array of a users sites. So added site meta to the queue here, might have some database queries for some developers. We can do this in another change, but I wanted to have one place in core where use this new functionality. It helped me test it.

File: tests/phpunit/tests/multisite/siteMeta.php (hunk 1 above)
Comment (spacedmonkey) [reply tests/phpunit/tests/multisite/siteMeta.php]:
```suggestion
			$this->assertSameSets( $site_ids, array( self::$site_id, self::$site_id2 ), 'This should have two sites meta' );
```

**Review by mukeshpanchal27 — APPROVED** [review]
Thanks @spacedmonkey, LGTM. 

The PR shows two error.
1. Unit test error in PHP 8.0 multisite because of timeout.
2. I noticed that there are unchanged files with check annotations, which is a bit strange because they don't appear in the changed files. 🤔

File: src/wp-includes/ms-site.php [hunk 3]
```diff
@@ -354,8 +355,28 @@ function _prime_site_caches( $ids, $update_meta_cache = true ) {
...
+		update_site_cache( $fresh_sites, false );
 	}
+
+	if ( $update_meta_cache ) {
+		wp_lazyload_site_meta( $ids );
+	}
+}
+
+
+/**
```
Comment (peterwilsoncc) [inline src/wp-includes/ms-site.php]:
```suggestion
}

/**
```

**Review by peterwilsoncc — APPROVED** [review]
Looks good for commit following the PR switching to `get_num_queries()`.

**Comment by spacedmonkey** [comment]
Committed. https://github.com/WordPress/wordpress-develop/commit/1a5b52a17e0bfa1ca2d9133466b5ead098e3a1ba
